### PR TITLE
Shadow Props supported on ReactNativeSkia and RSKDrawUtils refactored

### PR DESCRIPTION
**Pull request by shefayaparwin**
1. Supported Shadow props on React Native Skia. Props handled (shadowColor, shadowOffset, shadowOpacity, shadowRadius)
2. Refactored RSKDrawUtil to decouple it from RSKComponents.
3. Handled default value for background, border & shadow color as black.

### CODE REVIEW

File: ReactSkia/views/common/RSkDrawUtils.cpp [hunk 1]
```diff
@@ -7,111 +7,189 @@
...
+      Float endRadius{0};
+      Float width;
+      Float angle;
+  };
+void setPathEffect(BorderStyle borderStyle,int strokeWidth,SkPaint *paint)
 {
-
-    if(isDrawVisible(backgroundColor,opacity)){
-         drawRect(DrawFillRect,canvas,frame,borderProps,backgroundColor,opacity);
+    float Dash_interval[]={strokeWidth,strokeWidth/2};
```
Comment (pprabhu-ng) [inline ReactSkia/views/common/RSkDrawUtils.cpp]: variable naming in camelcase format ?

File: ReactSkia/views/common/RSkDrawUtils.cpp [hunk 2]
```diff
@@ -7,111 +7,189 @@
...
 
+    bool verticalEdge=(((borderEdge == LeftEdge) || (borderEdge == RightEdge)) ? true : false);
+    bool growCW=(((borderEdge == TopEdge) ||(borderEdge == RightEdge)) ? true : false);
+    bool cornerOnRightEdge=(((borderEdge == BottomEdge) ||(borderEdge== RightEdge)||(borderEdge == TopEdge)) ? true : false);
+    bool cornerOnBottomEdge=(((borderEdge == LeftEdge) ||(borderEdge == BottomEdge ) ||(borderEdge == RightEdge)) ? true : false);
+
+    auto outterSweepAngle =(growCW ? 45 : -45);
+    auto innerSweepAngle= (growCW ? -45 : +45);;
+
+    auto outterStartX= pathMetrics.outterStart.x ;
```
Comment (pprabhu-ng) [inline ReactSkia/views/common/RSkDrawUtils.cpp]: as an optimization ;than using two variables to store the x & y values can we use Point for having the x & y values 
Ex : outterStart = Point {pathMetrics.outterStart.x , pathMetrics.outterStart.y}

This way we get to manage only 4 variables here.

**Review by munezbn — APPROVED** [review]

File: ReactSkia/views/common/RSkDrawUtils.cpp (hunk 1 above)
Comment (shefayaparwin) [reply ReactSkia/views/common/RSkDrawUtils.cpp]: Will change it

File: ReactSkia/views/common/RSkDrawUtils.cpp (hunk 2 above)
Comment (shefayaparwin) [reply ReactSkia/views/common/RSkDrawUtils.cpp]: Good Point . Will address it.

**Review by pprabhu-ng — APPROVED** [review]

**Pull request merged**